common/web/gesture-recognizer/src/engine/headless/gestures/matchers/gestureMatcher.ts

```diff
@@ -5,22 +5,36 @@ import { GestureModel, GestureResolution, GestureResolutionSpec, RejectionDefaul
 import { ManagedPromise, TimeoutPromise } from "@keymanapp/web-utils";
 import { FulfillmentCause, PathMatcher } from "./pathMatcher.js";
 
+export interface PredecessorMatch<Type> {
+  readonly sources: GestureSource<Type>[];
+  readonly allSourceIds: string[];
+  readonly primaryPath: GestureSource<Type>;
+  readonly result: MatchResult<Type>;
+  readonly model?: GestureModel<Type>;
+  readonly baseItem: Type;
+}
+
 export interface MatchResult<Type> {
-  matched: boolean,
-  action: GestureResolution<Type>
+  readonly matched: boolean,
+  readonly action: GestureResolution<Type>
 }
 
-export interface MatchResultSpec {
-  matched: boolean,
-  action: GestureResolutionSpec
+export interface MatchResultSpec<Type> {
```

Comment:
Why does this need `<Type>` given `<Type>` is not used anywhere in the interface?